HW-2 Reposotory Pattern적용

jnugg0819 opened the pull request:
- repository 패턴 적용으로 Business로직단에는 Remote, Local구분없이 사용할 수 있도록 조치
- 필요없는 주석 제거

jnugg0819 committed:
HW-2 Reposotory Pattern적용

- repository 패턴 적용으로 Business로직단에는 Remote, Local구분없이 사용할 수 있도록 조치
- 필요없는 주석 제거

mingue0605 reviewed (approved):
리뷰달것도없네.. ㅇㅅㅇ

realizer12 commented on presentation/src/main/java/org/techtown/presentation/feature/main/NewsDetailFragment.kt:
```diff
@@ -58,7 +60,7 @@ class NewsDetailFragment : BaseFragment<FragmentNewsDetailBinding>(R.layout.frag
 
         CoroutineScope(Dispatchers.IO).launch {
-            newsRepository.getAllSavedArticles().collect { data ->
+            newsRepository.getAllArticles().collect { data ->
```
상미님 여기는 이미  datatsource에서 io로직들에 대하여,  
flowOn으로  io Dispatcher 를 적용해줬기 때문에 호출하는 쪽에서는  io를 해줄필요가 없을것 같아엽

여기를 viewLifecycleOwner.lifecycleScope 나 viewLifecycleOwner.repeatlifecycleScope
로 코루틴 호출을 해주고, 아래 Withcontext dispacture.Main은 없애주는건 어떤가엽??

jnugg0819 replied:
저거 근데 코루틴 스콮으로 안넣어주면 에러 나던데 음 lifecycleScope으로 하면 안나려나 함해볼게여

realizer12 replied:
오까이

jnugg0819 replied:
821ec1747235c266f1b6776406851ad4e9ff365c

realizer12 commented on presentation/src/main/java/org/techtown/presentation/datasource/local/LocalDataSourceImpl.kt:
```diff
@@ -0,0 +1,33 @@
+
+    override fun addArticle(articles: Articles, callback: () -> Unit) {
+        appDatabase.articleDao().insertArticle(articles)
+        callback()
```
상미님 여기도 insert랑 delete 도 coroutine 으로 바꾸는게 좋을것 같아엽.
suspend 랑 flowon으로 io 주고  호출하는 곳에서는  lifecycleScope로 처리해서 
suspend쪽  io처리 다 하고 돌아와서 동기로 진행되니까 
딱히 callback 안넣어줘도 될거 같고엽~

jnugg0819 replied:
995a39c83aa255a6b18c38889727d60cda363ba4

realizer12 commented on presentation/src/main/java/org/techtown/presentation/repository/NewsRepositoryImpl.kt:
```diff
@@ -1,60 +1,43 @@
-        appDatabase.articleDao().insertArticle(articles)
-        callback()
+        localDataSource.addArticle(articles = articles) {
+            callback()
```
요기 위 멘트 처럼  코루틴 처리로 해주세엽~
그리고 dao쪽도 바꿔야할것 같아엽

jnugg0819 replied:
이것 위에 커밋이랑 같이올렸슴다

jnugg0819 committed:
HW-2 중복 Dispatcher제거하고 lifecycleScope으로 통일

jnugg0819 committed:
HW-2 DAO함수들 suspend로 변경

- 연속적으로 가져올 데이터가 없으니까 Flow안씀
- callback제거
- CoroutineScope -> ViewLifecycle Scope으로 변경

realizer12 reviewed (approved):
나이스
